homeassistant/components/media_player/snapcast.py

```diff
@@ -0,0 +1,88 @@
+"""
+homeassistant.components.media_player.snapcast
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Provides functionality to interact with Snapcast clients.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.snapcast/
+"""
+
+import logging
+
+from homeassistant.const import (
+    STATE_ON, STATE_OFF)
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice,
+    SUPPORT_VOLUME_SET, SUPPORT_VOLUME_MUTE)
+
+SUPPORT_SNAPCAST = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE
+DOMAIN = 'snapcast'
+REQUIREMENTS = ['snapcast==1.0.0']
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the Snapcast platform. """
+    import snapcast
+    import socket
+    host = config.get('host')
+    port = config.get('port', snapcast.CONTROL_PORT)
+    if not host:
+        _LOGGER.error('No snapserver host specified')
+        return
+    try:
+        server = snapcast.Snapserver(host, port)
+        add_devices([SnapcastDevice(client) for client in server.clients])
```

Comment:
This doesn't look like it should be inside this try block. If this line were to throw a socket.gaierror, the log would show a problem with the server. If this line can't throw a socket.gaierror, there is no reason to handle the possible exception. I would this:

```python
try:
    server = snapcast.Snapserver(host, port)
except socket.gaierror:
    _LOGGER.error('Could not connect to Snapcast server at %s:%d', host, port)
    return

add_devices([SnapcastDevice(client) for client in server.clients])
```